### DISCUSSION TITLE
Need to Send OTP to User Email on Direct Grand flow

### DISCUSSION THREAD

**Discussion by ganapathy8974** [keycloak/keycloak#27598] [Q&A]
Hello,
I need to send an OTP email while calling this **_/protocol/openid-connect/token_** direct grand endpoint without issuing the new token. While I am sending the totp parameter, if it's valid, this API will share the token. 

1. Is there any spi/plugin available?
2. If you have any ideas on this, kindly share your thoughts as well.

**Comment by darius-m** [comment]
Would your use-case be covered by the device authorization grant, perhaps?

**Reply by ganapathy8974** [reply]
I would like to trigger OTP to user email, based on their credentials
